stdlib/internal/src/main/java/org/ballerinalang/stdlib/internal/execballerina/ExecBallerinaDoc.java

```diff
@@ -0,0 +1,316 @@
+            
+            Process process;
+            BufferedReader reader = null;
+            BufferedReader readerEr = null;
+            try {
+                // Running the command
+                StringBuilder output = new StringBuilder();
+                process = Runtime.getRuntime().exec(balCommand);
+                process.waitFor();
+                reader = new BufferedReader(new InputStreamReader(process.getInputStream(), StandardCharsets
+                        .UTF_8));
+                readerEr = new BufferedReader(new InputStreamReader(process.getErrorStream(), StandardCharsets
+                        .UTF_8));
+                
+                String lineEr;
+                while ((lineEr = readerEr.readLine()) != null) {
+                    output.append(lineEr).append("\n");
+                }
+                
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    output.append(line).append("\n");
+                }
+                
+                String adjusted = output.toString().replaceAll("(?m)^[ \t]*\r?\n", "");
+                context.setReturnValues(new BString(adjusted));
+            } catch (InterruptedException e) {
+                String msg = "Error occurred while waiting for ballerina command to finish: " + e.getMessage();
+                log.error(msg);
+                context.setReturnValues(BLangVMErrors.createError(context, msg));
+                throw new BallerinaException(e.getMessage());
+            } catch (IOException e) {
+                String msg = "Error occurred executing command or reading out of the command: " + e.getMessage();
+                log.error(msg);
+                context.setReturnValues(BLangVMErrors.createError(context, msg));
+            } finally {
+                try {
+                    if (reader != null) {
+                        reader.close();
+                    }
+                } catch (IOException e) {
+                    String msg = "Error occurred closing standard output: " + e.getMessage();
+                    log.error(msg);
+                    context.setReturnValues(BLangVMErrors.createError(context, msg));
+                }
+                try {
+                    if (readerEr != null) {
+                        readerEr.close();
+                    }
+                } catch (IOException e) {
+                    String msg = "Error occurred closing standard error output: " + e.getMessage();
+                    log.error(msg);
+                    context.setReturnValues(BLangVMErrors.createError(context, msg));
+                }
+            }
+        }
+    }
+    
+    /**
+     * Build the command to pass to ballerina doc command.
+     * @param context The function context.
+     * @return The command.
+     */
+    private static String buildCommand(Context context) {
```

Comment:
this function is too long and deep, let's break it up

Reply:
Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/9227/commits/47175543bf31fffd521f8b6d5d61d736aefce8ce